SpokeStack/SpokeStack/GoogleSpeechRecognizer.swift

```diff
@@ -0,0 +1,188 @@
+                                                                        finished = true
+                                                                    }
+                                                                    
+                                                                    if finished {
+                                                                        
+                                                                        let context: SPSpeechContext = SPSpeechContext(transcript: alt.transcript, confidence: alt.confidence)
+                                                                        
+                                                                        strongSelf.delegate?.didRecognize(context)
+                                                                        strongSelf.delegate?.didFinish()
+                                                                        strongSelf.stopStreaming()
+                                                                    }
+                                                                }
+                                                                
+            })
+            
+            /// authenticate using an API key obtained from the Google Cloud Console
+            
+            self.call.requestHeaders.setObject(NSString(string: self.googleConfiguration.apiKey),
+                                               forKey:NSString(string:"X-Goog-Api-Key"))
+            
+            /// if the API key has a bundle ID restriction, specify the bundle ID like this
+            
+            self.call.requestHeaders.setObject(NSString(string:Bundle.main.bundleIdentifier!),
+                                               forKey:NSString(string:"X-Ios-Bundle-Identifier"))
+            
+            self.call.start()
+            self.streaming = true
+            
+            /// send an initial request message to configure the service
+            
+            self.writer.writeValue(self.streamingRecognizerRequest)
+        }
+        
+        /// send a request message containing the audio data
+        
+        let streamingRecognizeRequest: StreamingRecognizeRequest = StreamingRecognizeRequest()
+        streamingRecognizeRequest.audioContent = audioData as Data
+        
+        self.writer.writeValue(streamingRecognizeRequest)
+    }
+}
+
+extension GoogleSpeechRecognizer: AudioControllerDelegate {
+    
+    func setupFailed(_ error: String) {
+        
+        self.streaming = false
+        self.delegate?.didFinish()
+    }
+    
+    func processSampleData(_ data: Data) -> Void {
+        
+        /// Convert to model and pass back to delegate
+        
+        self.audioData.append(data)
+        
+        /// We recommend sending samples in 100ms chunks
+        
+        let chunkSize: Int = Int(0.1 * Double(AudioController.shared.sampleRate) * 2)
```

Comment:
I think this results in `16000*0.1*2=3200` so  `0.0125`ms chunks? My math must be wrong somehow.

Reply:
actually, the `2` is assuming that the input is stereo.